File: Pset6/sentimental-readability/readability.py
from cs50 import get_string


def main():
    text = get_string("Text: ")
    letters = 0
    words = 1
    sentences = 0

    for c in range(len(text)):
        if text[c].isalpha():
            letters += 1

        if text[c] == " ":
            words += 1

        if text[c] == '.' or text[c] == '!' or text[c] == '?':
            sentences += 1

    l = (letters / words) * 100
    s = (sentences / words) * 100
    index = 0.0588 * l - 0.296 * s - 15.8
    index = round(index)

    if index < 1:
        print("Before Grade 1")

    elif index > 16:
        print("Grade 16+")

    else:
        print(f"Grade {index}")


main()

File: Pset2/readability/readability.c
#include <cs50.h>
#include <ctype.h>
#include <math.h>
#include <stdio.h>
#include <string.h>

int main(void)
{
    // Prompt the user for some text
    string text = get_string("text: ");
    int letters = 0;
    int words = 1;
    int sentences = 0;

    // Count the number of letters, words, and sentences in the text
    for (int i = 0, length = strlen(text); i < length; i++)
    {
        if (isalpha(text[i]) != 0)
        {
            letters++;
        }
        if ((text[i]) == ' ')
        {
            words++;
        }
        if ((text[i]) == '.' || text[i] == '!' || text[i] == '?')
        {
            sentences++;
        }
    }

    // Compute the Coleman-Liau index
    float L = ((float) letters / words) * 100;
    float S = ((float) sentences / words) * 100;
    float index = 0.0588 * L - 0.296 * S - 15.8;
    index = round(index);

    // Print the grade level
    if (index < 1)
        printf("Before Grade 1\n");
    else if (index > 16)
        printf("Grade 16+\n");
    else
        printf("Grade %i\n", (int) index);
}
